### PR TITLE
documentation: Guild members

### PR DESCRIPTION
**Pull request by CoocooFroggy**
# Description

Improves documentation of guild member related endpoints. Changes `IGuild#searchMembersGateway` behavior to be consistent with documentation and `IGuild#searchMembers`.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] Ran `dart analyze` or `make analyze` and fixed all issues
- [x] Ran `dart format --set-exit-if-changed -l 160 ./lib` or `make format` and fixed all issues
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked my changes haven't lowered code coverage

### CODE REVIEW

File: lib/src/core/guild/guild.dart [hunk 1]
```diff
@@ -309,9 +309,9 @@ abstract class IGuild implements SnowflakeEntity {
   /// Fetches member from API
   Future<IMember> fetchMember(Snowflake memberId);
 
-  /// Allows to fetch guild members. In future will be restricted with `Privileged Intents`.
+  /// Allows to fetch guild members. Restricted with [GatewayIntents.guildMembers].
   /// [after] is used to continue from specified user id.
-  /// By default limits to one user - use [limit] parameter to change that behavior.
+  /// By default limits to one user—use [limit] parameter to change that behavior.
```
Comment (abitofevrything) [inline lib/src/core/guild/guild.dart:314]: This shouldn't be changed - the hyphen is used to indicate a change of topic just like in this sentence. Removing the space around it makes it seem like `user-use` is a hyphenated word, which it isn't.

**Review by abitofevrything — CHANGES_REQUESTED** [review]

File: lib/src/core/guild/guild.dart (hunk 1 above)
Comment (CoocooFroggy) [reply lib/src/core/guild/guild.dart:314]: Ah I actually changed the hyphen `-` to an em dash `—`. A hyphen is used to combine two words, while an em dash indicates the change in topic.

Note that there are never spaces before or after either of these punctuation marks.

> Hyphens (-) are used to connect two or more words (and numbers) into a single concept, especially for building adjectives. 

https://owl.purdue.edu/owl/general_writing/punctuation/hyphen_use.html

> Dashes are used to set off or emphasize the content enclosed within dashes or the content that follows a dash.

https://owl.purdue.edu/owl/general_writing/punctuation/punctuation/index.html

Comment (CoocooFroggy) [reply lib/src/core/guild/guild.dart:314]: It's tough to see with a monospace font, but it's more obvious in viewing documentation.

doc

Comment (abitofevrything) [reply lib/src/core/guild/guild.dart:314]: Right. I'd rather keep this in simple ASCII characters though, where "internet convention" says we use the spacing to indicate the change in topic.

In new documentation I wouldn't mind this, but I see no point in changing preexisting documentation for this.

Comment (abitofevrything) [reply lib/src/core/guild/guild.dart:314]: Though this is only documentation, so I won't complain too much if you'd rather keep it as is.

Comment (CoocooFroggy) [reply lib/src/core/guild/guild.dart:314]: Understandable. I think you're right in keeping formatting consistent.

Don't mind me if I make a pull request changing every hyphen to a dash in the future 😋

**Review by abitofevrything — APPROVED** [review]

**Pull request merged**